packages/components/src/navigation/menu/menu-title.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { Icon, search as searchIcon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Animate from '../../animate';
+import Button from '../../button';
+import MenuTitleSearch from './menu-title-search';
+import { MenuTitleHeadingUI, MenuTitleUI } from '../styles/navigation-styles';
+import { useNavigationMenuContext } from './context';
+
+export default function NavigationMenuTitle( {
+	hasSearch,
+	onSearch,
+	search,
+	title,
+} ) {
+	const [ isSearching, setIsSearching ] = useState( false );
+	const { menu } = useNavigationMenuContext();
+	const searchButtonRef = useRef();
+
+	if ( ! title ) {
+		return null;
+	}
+
+	const onCloseSearch = () => {
+		setIsSearching( false );
+
+		// Wait for the slide-in animation to complete before focusing the search button.
+		// eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+		setTimeout( () => {
+			searchButtonRef.current.focus();
+		}, 100 );
```

Comment:
This is very minor, but I see we are using this `100` magic number a few times. Maybe worth a constant?